vnext/Shared/WinRTWebSocketResource.cpp

```diff
@@ -8,6 +8,7 @@
 #include <Utilities.h>
 
 // Windows API
+#include <winrt/Windows.Foundation.Collections.h>
```

Comment:
why is this change needed?

Reply:
After the move to the shared project, it fails due to usage of winrt `IVector`.
```
D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(97,58): error C3779: 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collections::IVector<winrt::Windows::Security::Cryptography::Certificates::ChainValidatio
       nResult>,T>::Append': a function that returns 'auto' cannot be used before it is defined [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
                 with
                 [
                     T=winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult
                 ]
       D:\Extra\react\windows\fixcommon\vnext\build\x64\Debug\React.Windows.Desktop\Generated Files\winrt\impl\Windows.Foundation.Collections.0.h(432): message : see declaration of 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collectio
       ns::IVector<winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult>,T>::Append' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
                 with
                 [
                     T=winrt::Windows::Security::Cryptography::Certificates::ChainValidationResult
                 ]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(127,41): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Foundation::IAsyncAction' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(171,39): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Storage::Streams::DataWriterStoreOperation' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Des
       ktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(239,39): error C3312: no callable 'await_resume' function found for type 'winrt::Windows::Storage::Streams::DataWriterStoreOperation' [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Des
       ktop.vcxproj]
     4>D:\Extra\react\windows\fixcommon\vnext\Shared\WinRTWebSocketResource.cpp(373,23): error C3779: 'winrt::impl::consume_Windows_Foundation_Collections_IVector<winrt::Windows::Foundation::Collections::IVector<winrt::hstring>,T>::Append': a function that returns 'auto' cannot
       be used before it is defined [D:\Extra\react\windows\fixcommon\vnext\Desktop\React.Windows.Desktop.vcxproj]
```

Actually, what puzzles me is why it was NOT failing to compile before.

Reply:
@JunielKatarn The reason this is needed now is because you deleted the ReactWindowsCore/pch folder (there doesn't seem to be a Shared/pch folder anymore). The PCH used to bring along a bunch of WinRT headers, including the one you're having to include manually now.

I wonder if this is fine or if it will make it harder for WinUI 3... I suppose if the CI passes the WinUI 3 task then this should be ok.

Reply:
CI does pass WinUI 3.

Please note, there is a `Shared\Pch` directory:
```
 ls .\Shared\Pch\
Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---           5/26/2020  3:04 PM            118 pch.cpp
-a---           5/26/2020  3:04 PM            359 pch.h
```

However, being a shared project and not a concrete one (i.e. static lib), the PCH might be obtained form elsewhere.
In short, any critical includes should be added to either the Desktop VCXPROJ, MS.ReactNative VCXPROJ, or both.

This doesn't seem to be the case. The winrt Collections header is not being widely used.